### DISCUSSION TITLE
Custom shaders

### DISCUSSION THREAD

**Discussion by mirobartovic** [EmulatorJS/EmulatorJS#257] [Q&A] [answered]
Hello Ethan, thank you for the great work. I have a question regarding shaders. I didn't find a way to add new one. Is is possible by a user? Or do I need to open an issue, list the shaders I would like to add (some lightweight ones) and hope you will find some time to add the shader? I'm a C# and JS developer so maybe I could contribute to the project and do this update myself but I don't know where the shaders are hidden in the code. I'm a bit afraid it's really obvious and I somehow missed it 😁.

**Comment by ethanaobrien** [comment, marked as answer]
Hello! Its not too hard to add custom shaders. First make sure you have `EJS_DEBUG_XX` set to true so the js knows to pull the non-minified file (I recommend re-minifying it when putting it into production). 

To add new shaders from there:
1. On line `591` in the `emu-main.js` file is a json object containing the shader data, just add your shader to this object.
2. Then, on line `4369` in the `emu-main.js` file is another object containing the shader menu items, add your shader there and it should work

Let me know if you have trouble with anything or have any questions!

**Reply by mirobartovic** [reply]
Thanks, looks simple. For some reason I was previously looking through all the folders looking for some shader file and not in the main code itself.

**Reply by ethanaobrien** [reply]
No problem! If you have more shaders to add you can make a pr. Id gladly accept it!

**Reply by rpeveri** [reply]
could you add more shaders ? if so help me ? cheers
